### PR TITLE
try/catch around setItem

### PR DESCRIPTION
**Pull request by triptec**
Safari in private browsing throws quota exceeded then one is using setItem on sessionstorage and localstorage, this is to catch that.

More info http://stackoverflow.com/questions/14555347/html5-localstorage-error-with-safari-quota-exceeded-err-dom-exception-22-an

### CODE REVIEW

**Comment by triptec** [comment]
hold off on this, looking at the polyfill for sessionStorage

**Comment by jdegoes** [comment]
It might just be better to try/catch in the event tracker. This way, the event tracker won't ever throw exceptions, regardless of the underlying failure -- I assume the goal is that, if the event tracker fails, it should not affect the code that is calling the track function?

**Comment by triptec** [comment]
But isn't this called by the event listeners that are setup? And the sessionStorage polyfill does this for sessionStorage but not localStorage so sessionStorage errors are already caught with catch(o_0){//iOS private browsing} if I'm not missinformed

**Comment by jdegoes** [comment]
Well, first, what is the goal? To avoid crashing client code with queued events?

**Comment by triptec** [comment]
Yes the goal is to avoid crashing, but it's crashing when trying to scribe_vid in visitorId() where it runs localStorage.setItem('scribe_vid', ...) so I'm not sure it has to do with cached events

File: src/scribe-analytics.js [hunk 1]
```diff
@@ -174,6 +174,33 @@ if (typeof Scribe === 'undefined') {
...
+        return storage.getItem(key);
+      },
+      setItem: function(storage, key, value) {
+        try {
+          return storage.setItem(key, value);
+        } catch(domException) {
+          if (domException.name === 'QuotaExceededError' ||
+              domException.name === 'SecurityError' ||
+              domException.name === 'NS_ERROR_DOM_QUOTA_REACHED') {
+            return null
```
Comment (jdegoes) [inline src/scribe-analytics.js]: Missing semi-colon. Also, what does the spec say it will return?

File: src/scribe-analytics.js [hunk 2]
```diff
@@ -1067,23 +1094,23 @@ if (typeof Scribe === 'undefined') {
       this.context.fingerprint = Env.getFingerprint();
 
       this.context.sessionId = (function() {
-        var sessionId = sessionStorage.getItem('scribe_sid') || Util.genGuid();
+        var sessionId = Util.store.session.getItem('scribe_sid') || Util.genGuid();
 
-        sessionStorage.setItem('scribe_sid', sessionId);
+        Util.store.session.setItem('scribe_sid', sessionId);
```
Comment (jdegoes) [inline src/scribe-analytics.js:1099]: See, another option may just be try / catching this. It will be less code (= smaller distribution size).

**Pull request closed without merging**